quickstart/aspnet-core/ContosoTeamStats/RedisConnection.cs

```diff
@@ -142,21 +142,26 @@ private async Task ForceReconnectAsync(bool initializing = false)
                 _firstErrorTime = DateTimeOffset.MinValue;
                 _previousErrorTime = DateTimeOffset.MinValue;
 
-                if (_connection != null)
+                // Create a new connection
+                ConnectionMultiplexer _newConnection = await ConnectionMultiplexer.ConnectAsync(_connectionString);
+
+                // Swap current connection with the new connection
+                ConnectionMultiplexer _oldConnection = Interlocked.Exchange(ref _connection, _newConnection);
+
+                if (_oldConnection != null)
                 {
                     try
                     {
-                        await _connection.CloseAsync();
+                        await _oldConnection.CloseAsync();
                     }
                     catch
                     {
                         // Ignore any errors from the old connection
                     }
                 }
 
-                Interlocked.Exchange(ref _connection, null);
-                ConnectionMultiplexer newConnection = await ConnectionMultiplexer.ConnectAsync(_connectionString);
-                Interlocked.Exchange(ref _connection, newConnection);
+                // Dispose old connection
+                Interlocked.Exchange(ref _oldConnection, null);
```

Comment:
One other note is it might be valuable to create a new issue along the way of merging this PR and closing #38 to capture and track the last point from #38 that isn't yet resolved by this PR:

> 3\. Dispose old connection - **this one can still cause problems to any thread still using the connection, not sure what to do with it.**

(emphasis added)